workers/unity/Packages/io.improbable.worker.sdk/IOStream.cs

```diff
@@ -0,0 +1,120 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using Improbable.Worker.CInterop.Internal;
+
+namespace Improbable.Worker.CInterop
+{
+    public unsafe class IOStream : IDisposable
+    {
+        private readonly CIO.StreamHandle stream;
+
+        private IOStream(CIO.StreamHandle stream)
+        {
+            this.stream = stream;
+        }
+
+        /// <inheritdoc cref="IDisposable"/>
+        public void Dispose()
+        {
+            stream.Dispose();
+            GC.SuppressFinalize(this);
+        }
+
+        public static IOStream CreateRingBufferStream(uint capacity)
+        {
+            return new IOStream(CIO.CreateRingBufferStream(capacity));
+        }
+
+        public static IOStream CreateFileStream(string fileName)
```

Comment:
This should probably take the 'OpenMode' as a parameter to avoid future breaking changes. 

You may need to redefine the enum here to avoid leaking the internal namespace. (I'd see what the Worker SDK currently does in these kinds of cases, like `Authority`)